pyannotate_runtime/collect_types.py

```diff
@@ -766,10 +777,12 @@ def _trace_dispatch(frame, event, arg):
                 resolved_types = prep_args(arg_info)
                 _task_queue.put(KeyAndTypes(function_key, resolved_types))
             elif event == 'return':
-                # This event is also triggered if a function raises an exception,
-                # and in this case the return value is 'None'.  There doesn't seem
-                # to be a way to distinguish an exception from a None return,
-                # unfortunately.
+                # This event is also triggered if a function raises an exception.
+                # We can tell the difference by looking at the bytecode.
+                # TODO: Also recognize YIELD_VALUE opcode.
```

Comment:
Maybe mention in the comment that profiling hooks aren't invoked for C functions so the bytecode always exists.